querybook/webapp/components/Survey/Survey.tsx

```diff
@@ -0,0 +1,182 @@
+import clsx from 'clsx';
+import React from 'react';
+import toast, { Toast } from 'react-hot-toast';
+
+import {
+    ISurvey,
+    IUpdateSurveyFormData,
+    SurveySurfaceType,
+    SurveyTypeToQuestion,
+} from 'const/survey';
+import { saveSurveyRespondRecord } from 'lib/survey/triggerLogic';
+import { SurveyResource } from 'resource/survey';
+import { TextButton } from 'ui/Button/Button';
+import { ResizableTextArea } from 'ui/ResizableTextArea/ResizableTextArea';
+
+import { StarRating } from './StarRating';
+
+import './Survey.scss';
+
+export interface ISurveyProps {
+    surface: SurveySurfaceType;
+    surfaceMeta?: Record<string, any>;
+    toastProps: Toast;
+}
+
+export const Survey: React.FC<ISurveyProps> = ({
+    surface,
+    surfaceMeta = {},
+    toastProps,
+}) => {
+    const [survey, setSurvey] = React.useState<ISurvey | null>(null);
+    const handleDismiss = React.useCallback(() => {
+        toast.dismiss(toastProps.id);
+    }, [toastProps.id]);
+
+    const formDOM = !survey ? (
+        <SurveyCreationForm
+            surface={surface}
+            surfaceMeta={surfaceMeta}
+            onSurveyCreation={setSurvey}
+            onDismiss={handleDismiss}
+        />
+    ) : (
+        <SurveyUpdateForm
+            surface={surface}
+            survey={survey}
+            surfaceMeta={surfaceMeta}
+            onSurveyUpdate={setSurvey}
+            onDismiss={handleDismiss}
+        />
+    );
+
+    return (
+        <div
+            className={clsx(
+                'Survey',
+                toastProps.visible ? 'Survey-enter' : 'Survey-leave'
+            )}
+        >
+            {formDOM}
+        </div>
+    );
+};
+
+function getSurveyQuestion(
```

Comment:
can you add some comment with example for this function. it appears like magic to me at first glance.

Reply:
updated